### ISSUE TITLE
volume thicknessFactor bug?

### ISSUE THREAD

**Issue by echadwick-wayfair** [BabylonJS/Babylon.js#10193]
![image](https://user-images.githubusercontent.com/40177036/114784240-f8a98f00-9d48-11eb-95cf-b0a969d3837b.png)

KHR_materials_volume thicknessFactor seems to be ignored in the Sandbox. 

There is no thicknessTexture in this example, not sure if that affects it, or if it's the cause.

[volumeThicknessFactor.zip](https://github.com/BabylonJS/Babylon.js/files/6313879/volumeThicknessFactor.zip)

Thanks!

**Comment by echadwick-wayfair** [comment]
It was mentioned in this issue in the glTF-Sample-Viewer repo
https://github.com/KhronosGroup/glTF-Sample-Viewer/issues/302

**Comment by Popov72** [comment]
The production site is not updated yet with the latest changes regarding the thickness factor of the volume extension. When it is, [this link](https://sandbox.babylonjs.com/?assetUrl=https://assets.babylonjs.com/meshes/KHR_materials_volume_testing.glb) should render:
![image](https://user-images.githubusercontent.com/4152247/114861161-2e07b880-9ded-11eb-80a3-95ccdb5d7524.png)
and not:
![image](https://user-images.githubusercontent.com/4152247/114861224-44157900-9ded-11eb-8323-b4de13503e84.png)

However, even with those changes, your asset will render as:
![image](https://user-images.githubusercontent.com/4152247/114861298-5a233980-9ded-11eb-87be-db8be018e268.png)

The thickness only modulates the refraction transmittance in Babylon. The formula is:
```javascript
refractionTransmittance *= exp(log(attenuationColor)*distance/attenuationDistance);
```
`distance = thicknessFactor` in the sample because there's no thickness texture and `attenuationDistance = +Infinity` because the gltf file does not provide a value. So, `exp(...) = (1,1,1)` and there's no change in the refraction transmittance color. Even if `attenuationDistance` is not +Infinity, `exp(...) = (1,1,1)` because `attenuationColor = (1,1,1)` (it's the default color when no value is provided in the gltf file).

I can see in the Khronos GLTF viewer that the IOR is modified by the thickness factor (or at least it's what it seems):
![image](https://user-images.githubusercontent.com/4152247/114863898-a02dcc80-9df0-11eb-9e46-309a355c2842.png)

From my understanding of [the spec](https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_materials_volume), IOR should not be modified by the volume extension...

**Comment by sebavan** [comment]
Adding @bghgary about the spec.

**Comment by echadwick-wayfair** [comment]
Apparently [they are not affecting the IOR at all](https://github.com/KhronosGroup/glTF-Sample-Viewer/issues/302#issuecomment-820460148), rather they are using thickness value to determine the length of the ray in the volume, therefore higher values can lead to a larger offset of the ray.

**Comment by Popov72** [comment]
I have answered in the issue: I think they are recomputing a ray at the exit point of the surface, something we don't do in Babylon, hence the difference.

**Comment by Popov72** [comment]
We have added a new flag to use the thickness to modulate the ray length in the refraction computation: https://github.com/BabylonJS/Babylon.js/pull/10212

With this flag, the output of volumeThicknessFactor.glb is:
![image](https://user-images.githubusercontent.com/4152247/115118603-453cd680-9fa4-11eb-81b0-f5255a46374b.png)

**Comment by sebavan** [comment]
closed by #10212
